lib/puppet/environments.rb

```diff
@@ -379,6 +379,7 @@ def get(name)
       # This strategy favors smaller memory footprint over environment
       # retrieval time.
       clear_all_expired
+      name = name.to_sym
```

Comment:
I think [`Cached#clear`](https://github.com/puppetlabs/puppet/blob/255c4422f2c1df98d20a1257c804cbef04e027d2/lib/puppet/environments.rb#L415-L416) and [`Cached#get_conf`](https://github.com/puppetlabs/puppet/blob/255c4422f2c1df98d20a1257c804cbef04e027d2/lib/puppet/environments.rb#L464-L465) need similar treatment.